test/ui-testing/publishButton.test.js

```diff
@@ -1,14 +1,24 @@
 const timeout = process.env.SLOWMO ? 60000 : 10000;
 const fs = require('fs');
-beforeAll(async () => {
-  path = fs.realpathSync('file://../examples/index.html');
-  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
-});
 
 describe('Publish button', () => {
-  
-  test('something we are testing described here', () => {
+  var $;
+  beforeAll(async () => {
+    path = fs.realpathSync('file://../examples/index.html');
+    await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
+
+    const {JSDOM} = require("jsdom");
+    const myJSDom = new JSDOM(fs.readFileSync(path));
+    $ = require('jquery')(myJSDom.window);
+  });
 
+  test('something we are testing described here', () => {
+    // Check initially that Publish button is disabled.
+    expect($('.ple-publish').prop('disabled')).toBe(true);
```

Comment:
OK this is what's failing now. And it looks like a real failure - but is it? In theory, the editor's Publish button should really be disabled on initial page load, right? because it's waiting for you to input a title and some text before it thinks you should be able to press Publish. 

Maybe we should start by checking something more basic, like this?

`expect($('.ple-publish')).toBeDefined();`

That way we know we're looking at the correct page at least, then we can try to figure out why the button is not disabled. How does that sound?

Reply:
I'm working from this list of Jest assertions: https://jestjs.io/docs/using-matchers

Reply:
Ah, i see. From your gitter chat comment, you're saying the button gets created, but validation is never run (`_editor.validate()`), so it never adds the `disabled` state to the button because it never checks if you've added a title. 

That would be good, in that we are reading the actual HTML. But why isn't it running validation? Normally these lines should "boot up" the editor including running initial validations:

https://github.com/publiclab/PublicLab.Editor/blob/f8e1edb6ff992a03aa593feae53783f2b9867bd2/examples/index.html#L410-L419

And after all, the test file this one is based on does the exact same thing - and relies on that Editor "boot-up" to have run: https://github.com/publiclab/PublicLab.Editor/blob/main/spec/javascripts/button_spec.js

How can we see the errors that may occur preventing that bootup from happening? When we look through the log in this PR, are there signs that some code hit an error or did not run?

Earlier tests seem to have run. Did they load the page properly?https://github.com/publiclab/PublicLab.Editor/runs/4510360996?check_suite_focus=true#step:7:8

Hmm, a lot of them seem to be asynchronous, meaning they're waiting for other stuff to complete before trying to run. Do we need to do the same for this new test? 

https://github.com/publiclab/PublicLab.Editor/blob/f8e1edb6ff992a03aa593feae53783f2b9867bd2/test/ui-testing/CustomInsert.test.js#L9-L13

Reply:
So would that be like this, plus adding `async` to the top of the containing function?

```suggestion
    await expect($('.ple-publish').prop('disabled')).toBe(true);
```